sqlite/src/func.c

```diff
@@ -796,11 +798,17 @@ static void comdb2CtxinfoFunc(
   if( sqlite3_value_type(argv[0])!=SQLITE_TEXT ){
     return;
   }
+
+  struct sql_thread *thd = pthread_getspecific(query_info_key);
+  struct sqlclntstate *clnt = thd!=NULL ? thd->clnt : NULL;
+
   zName = (const char *)sqlite3_value_text(argv[0]);
   if( sqlite3_stricmp(zName, "parallel")==0 ){
-    struct sql_thread *thd = pthread_getspecific(query_info_key);
-    struct sqlclntstate *clnt = thd!=NULL ? thd->clnt : NULL;
     sqlite3_result_int(context, clnt!=NULL && clnt->conns!=NULL);
+  } else if( sqlite3_stricmp(zName, "user")==0 ){
+    if (clnt) {
+      sqlite3_result_text(context, get_current_user(clnt), -1, SQLITE_STATIC);
```

Comment:
What if get_current_user returns null?

Reply:
null = "no user", I have tests that covers the scenario.

Reply:
Ah, I see.  The original code returned an empty string if clnt was NULL; however, it returned NULL when get_current_user returned NULL, which would then translate to a SQL NULL.  Given that, my snippet above is unnecessary.